v1_21/src/main/java/com/denizenscript/denizen/nms/v1_21/impl/network/handlers/packet/DisguisePacketHandlers.java

```diff
@@ -114,25 +111,42 @@ public static ClientboundUpdateAttributesPacket processAttributesPacket(DenizenN
 
     public static ClientboundTeleportEntityPacket processTeleportPacket(DenizenNetworkManagerImpl networkManager, ClientboundTeleportEntityPacket teleportEntityPacket, DisguiseCommand.TrackedDisguise disguise) throws IllegalAccessException {
         if (disguise.as.getBukkitEntityType() == EntityType.ENDER_DRAGON) {
-            ClientboundTeleportEntityPacket pNew = ClientboundTeleportEntityPacket.STREAM_CODEC.decode(DenizenNetworkManagerImpl.copyPacket(teleportEntityPacket, ClientboundTeleportEntityPacket.STREAM_CODEC));
-            TELEPORT_PACKET_YAW.setByte(pNew, EntityAttachmentHelper.adaptedCompressedAngle(teleportEntityPacket.getyRot(), 180));
-            return pNew;
+            PositionMoveRotation oldChange = teleportEntityPacket.change();
+            return new ClientboundTeleportEntityPacket(
+                    teleportEntityPacket.id(),
+                    new PositionMoveRotation(oldChange.position(), oldChange.deltaMovement(), oldChange.xRot(), EntityAttachmentHelper.adaptedAngle(oldChange.yRot(), 180)),
+                    teleportEntityPacket.relatives(),
+                    teleportEntityPacket.onGround()
+            );
         }
         return sendDisguiseForPacket(networkManager, teleportEntityPacket, disguise);
     }
 
 
     public static ClientboundMoveEntityPacket.Rot processMoveEntityRotPacket(DenizenNetworkManagerImpl networkManager, ClientboundMoveEntityPacket.Rot rotPacket, DisguiseCommand.TrackedDisguise disguise) {
         if (disguise.as.getBukkitEntityType() == EntityType.ENDER_DRAGON) {
-            return new ClientboundMoveEntityPacket.Rot(disguise.entity.getBukkitEntity().getEntityId(), EntityAttachmentHelper.adaptedCompressedAngle(rotPacket.getyRot(), 180), rotPacket.getxRot(), rotPacket.isOnGround());
+            return new ClientboundMoveEntityPacket.Rot(
+                    disguise.entity.getBukkitEntity().getEntityId(),
+                    Mth.packDegrees(EntityAttachmentHelper.adaptedAngle(rotPacket.getyRot(), 180)),
+                    Mth.packDegrees(rotPacket.getxRot()),
```

Comment:
Why does the packet `get` in one format and `set` in the other format? that's jank